src/lib/types.ts

```diff
@@ -35,8 +35,27 @@ export interface Action {
    */
   cond?: (args: RunArgs) => boolean;
   run?: (args: RunArgs) => void;
+  /**
+   * Prevent children from being displayed at the root level of the palette.
+   *
+   * Default: `true`
+   */
+  isolateChildren?: boolean;
+}
+
+export interface ChildAction {
+  isolateChildren?: never;
 }
 
+export interface ParentAction {
+  parentActionId: never;
```

Comment:
chore: the types here are incorrect because a parent action itself can be child of another action. It's a hierarchical chain.

Example-

```md
                                                <- Toggle Personal Wi-Fi
                  <- Configure Personal Profile <- Toggle Personal Airplane Mode
Configure Profile
                  <- Configure Work Profile     <- Toggle Work Wi-Fi
                                                <- Toggle Work Airplane Mode

```

https://github.com/itaditya/solid-command-palette/blob/main/src/app/views/demo/NestedActionDemo/nestedActions.ts#L33-L53

Reply:
reverted to prior state @ https://github.com/itaditya/solid-command-palette/pull/15/commits/5d95e4f136ff94325e547076cb44f50adfce7fac